apps/src/templates/studioHomepages/teacherHomepageV2/TeacherHomepageDrawer.tsx

```diff
@@ -0,0 +1,247 @@
+    schoolName: existingSchoolInfo?.school_name,
+    schoolId: existingSchoolInfo?.school_id,
+    schoolZip: existingSchoolInfo?.school_zip,
+    schoolType: existingSchoolInfo?.school_type,
+  });
+
+  const tryUpdateSchoolInfo = async () => {
+    const hasNcesId =
+      schoolInfo.schoolId && !NON_SCHOOL_OPTIONS.includes(schoolInfo.schoolId);
+    analyticsReporter.sendEvent(
+      EVENTS.SCHOOL_INTERSTITIAL_SUBMIT,
+      {
+        hasNcesId: hasNcesId.toString(),
+        attempt: showSchoolInfoUnknownError ? 2 : 1,
+      },
+      PLATFORMS.BOTH
+    );
+    try {
+      await updateSchoolInfo({
+        schoolId: schoolInfo.schoolId,
+        country: schoolInfo.country,
+        schoolName: schoolInfo.schoolName,
+        schoolZip: schoolInfo.schoolZip,
+      });
+
+      analyticsReporter.sendEvent(
+        EVENTS.SCHOOL_INTERSTITIAL_SAVE_SUCCESS,
+        {
+          attempt: showSchoolInfoUnknownError ? 2 : 1,
+        },
+        PLATFORMS.BOTH
+      );
+
+      setSuccess(true);
+      setSchoolInfoInterstitialOpen(false);
+    } catch (error) {
+      analyticsReporter.sendEvent(
+        EVENTS.SCHOOL_INTERSTITIAL_SAVE_FAILURE,
+        {
+          attempt: showSchoolInfoUnknownError ? 2 : 1,
+        },
+        PLATFORMS.BOTH
+      );
+
+      if (!showSchoolInfoUnknownError) {
+        // First failure, display error message and give the teacher a chance
+        // to try again.
+        setShowSchoolInfoUnknownError(true);
+      } else {
+        // We already failed once, let's not block the teacher any longer.
+        onDrawerClose();
+      }
+    }
+  };
+
+  const handlePrimaryButtonClick = async () => {
+    if (success) {
+      // If the drawer is already showing success, we just want to close it.
+      onDrawerClose();
+    } else if (schoolInfoConfirmationOpen) {
+      // If the confirmation drawer is open, the user can click through to
+      // make the school info panel appear.
+      analyticsReporter.sendEvent(
+        EVENTS.CONFIRM_SCHOOL_CLICKED,
+        {},
+        PLATFORMS.BOTH
+      );
+      setSchoolInfoInterstitialOpen(true);
+      setSchoolInfoConfirmationOpen(false);
+    } else if (schoolInfoInterstitialOpen) {
+      // If the interstitial is open, we want to submit the school info.
+      tryUpdateSchoolInfo();
+    }
+  };
+
+  const onDrawerClose = () => {
+    if (schoolInfoInterstitialOpen) {
+      analyticsReporter.sendEvent(
+        EVENTS.SCHOOL_INTERSTITIAL_DISMISS,
+        {},
+        PLATFORMS.BOTH
+      );
+      setSchoolInfoInterstitialOpen(false);
+    } else if (schoolInfoConfirmationOpen) {
+      analyticsReporter.sendEvent(
+        EVENTS.UPDATE_SCHOOL_INFO_DIALOG_CLOSED,
+        {},
+        PLATFORMS.BOTH
+      );
+      setSchoolInfoConfirmationOpen(false);
+    } else if (success) {
+      setSuccess(false);
+    } else {
+      setSchoolInfoInterstitialOpen(false);
+      setSchoolInfoConfirmationOpen(false);
+      setSuccess(false);
```

Comment:
```suggestion
    }
    setSchoolInfoInterstitialOpen(false);
    setSchoolInfoConfirmationOpen(false);
    setSuccess(false);
```

I think we want to set these to false every time this code is run, not just if none of the above cases are true.